src/server/api/schema.js

```diff
@@ -622,6 +623,77 @@ const rootMutations = {
       const newCampaign = await campaignInstance.save()
       return editCampaign(newCampaign.id, campaign, loaders)
     },
+    copyCampaign: async (_, { id }, { user, loaders }) => {
+      const campaign = await loaders.campaign.load(id)
+      await accessRequired(user, campaign.organization_id, 'ADMIN')
+
+      const campaignInstance = new Campaign({
+        organization_id: campaign.organization_id,
+        title: 'COPY - ' + campaign.title,
+        description: campaign.description,
+        due_by: campaign.dueBy,
+        is_started: false,
+        is_archived: false
+      })
+      const newCampaign = await campaignInstance.save()
+      const newCampaignId = newCampaign.id
+      const oldCampaignId = campaign.id
+
+      let interactions = await r.knex('interaction_step')
+        .where({campaign_id: oldCampaignId })
+
+      const interactionsArr = []
```

Comment:
This looks correct.  It might be more concise in the form:
```
const interactionsArr = interactions.map( (interaction, index) => ({
  id: 'new'+interaction.id,
  questionText: interaction.question,
  ...
  parentInteractionId: ((interaction.parent_interaction_id) 
                                   ? 'new'+interaction.parent_interaction_id)
                                   : undefined)
}))
```